databricks/koalas/series.py

```diff
@@ -2440,6 +2440,76 @@ def sort_index(
         else:
             return first_series(kdf)
 
+    def swaplevel(self, i=-2, j=-1, copy=True) -> "Series":
+        """
+        Swap levels i and j in a MultiIndex.
+        Default is to swap the two innermost levels of the index.
+
+        Parameters
+        ----------
+        i, j : int, str
+            Level of the indices to be swapped. Can pass level name as string.
+        copy : bool, default True
+            Whether to copy underlying data. Must be True.
+
+        Returns
+        -------
+        Series
+            Series with levels swapped in MultiIndex.
+
+        Examples
+        --------
+        >>> midx = pd.MultiIndex.from_arrays([['a', 'b'], [1, 2]], names = ['word', 'number'])
+        >>> midx
+        MultiIndex([('a', 1),
+                    ('b', 2)],
+                   names=['word', 'number'])
+        >>> kser = ks.Series(['x', 'y'], index=midx)
+        >>> kser
+        word  number
+        a     1         x
+        b     2         y
+        dtype: object
+        >>> kser.swaplevel()
+        number  word
+        1       a       x
+        2       b       y
+        dtype: object
+        >>> kser.swaplevel(0, 1)
+        number  word
+        1       a       x
+        2       b       y
+        dtype: object
+        >>> kser.swaplevel('number', 'word')
+        number  word
+        1       a       x
+        2       b       y
+        dtype: object
+        """
+        assert copy
+
+        for index in (i, j):
+            if not isinstance(index, int) and index not in self.index.names:
+                raise KeyError("Level %s not found" % index)
+
+        i = i if isinstance(i, int) else self.index.names.index(i)
+        j = j if isinstance(j, int) else self.index.names.index(j)
+
+        for index in (i, j):
+            if index >= len(self.index.names) or index < -len(self.index.names):
+                raise IndexError(
+                    "Too many levels: Index of the series has only %s levels, "
+                    "%s is not a valid level number" % (len(self.index.names), index)
+                )
+
+        index_map = list(zip(self._internal.index_spark_column_names, self._internal.index_names))
+        index_map[i], index_map[j], = index_map[j], index_map[i]
+        index_spark_column_names, index_names = zip(*index_map)
+        internal = self._kdf._internal.copy(
```

Comment:
`self._internal` instead of`self._kdf._internal` ?

Reply:
May I ask what's their difference at usage?

I see `return self._kdf._internal.select_column(self._column_label)` is for `self._internal`.

Reply:
`self._kdf._internal` contains other columns from the anchor DataFrame.

E.g.,:

```py
>>> midx = pd.MultiIndex.from_arrays([['a', 'b'], [1, 2]], names = ['word', 'number'])
>>> kdf = ks.DataFrame({'a': [1, 2], 'b': ['x', 'y']}, index=midx)
>>> kdf
             a  b
word number
a    1       1  x
b    2       2  y
>>> kdf.b.swaplevel()
number  word
1       a       1
2       b       2
Name: a, dtype: int64
```

should be:

```py
>>> kdf.b.swaplevel()
number  word
1       a       x
2       b       y
Name: b, dtype: object
```

Reply:
Thanks for explaining!

I'm a little confused by "contains". Their `spark_frame` seems to be identical.
```
>>> kdf.b._internal.spark_frame.show()
+-----------------+-----------------+---+---+-----------------+
|__index_level_0__|__index_level_1__|  a|  b|__natural_order__|
+-----------------+-----------------+---+---+-----------------+
|                a|                1|  1|  x|      42949672960|
|                b|                2|  2|  y|      94489280512|
+-----------------+-----------------+---+---+-----------------+

>>> kdf.b._kdf._internal.spark_frame.show()
+-----------------+-----------------+---+---+-----------------+
|__index_level_0__|__index_level_1__|  a|  b|__natural_order__|
+-----------------+-----------------+---+---+-----------------+
|                a|                1|  1|  x|      42949672960|
|                b|                2|  2|  y|      94489280512|
+-----------------+-----------------+---+---+-----------------+

```

Reply:
I mean their metadata:

```py
>>> kdf.b._internal.column_labels
[('b',)]
>>> kdf.b._kdf._internal.column_labels
[('a',), ('b',)]
```

or 

```py
>>> kdf.b._internal.data_spark_columns
[Column<b'b'>]
>>> kdf.b._kdf._internal.data_spark_columns
[Column<b'a'>, Column<b'b'>]
```

Reply:
I see, that's clear to me now. Thank you!